spyder/plugins/editor/widgets/editor.py

```diff
@@ -2329,6 +2329,14 @@ def __refresh_readonly(self, index):
             if not osp.isfile(finfo.filename):
                 # This is an 'untitledX.py' file (newly created)
                 read_only = False
+            try:
+                # Try to open the file to see if the permissions allow writing

```

Comment:
Please change this to

     # Try to open the file to see if its permissions allow to write on it